src/lib.rs

```diff
@@ -260,6 +260,28 @@ impl JsValue {
     ///
     /// If this JS value is not an instance of a string or if it's not valid
     /// utf-8 then this returns `None`.
+    ///
+    /// # UTF-16 vs UTF-8
+    ///
+    /// Strings in JavaScript are encoded as UTF-16, but with one major
+    /// exception: they can contain unpaired surrogates. For some unicode
+    /// characters UTF-16 usese two 16-byte values.  These are called "surrogate
+    /// pairs" because they always come in pairs. In JavaScript, it is possible
+    /// for these surrogate pairs to be missing the other half, creating an
+    /// "unpaired surrogate".
+    ///
+    /// This method internally will use `TextEncoder` in JS to convert from
+    /// UTF-16 to UTF-8. This is normally perfectly fine...  unless there are
+    /// unpaired surrogates. In that case it will replace the unpaired
+    /// surrogates with U+FFFD (�, the replacement character). That means the
+    /// string returned from this function is actually different from the string
+    /// in JS!
+    ///
+    /// If you want to guarantee that the returned string is the same as the
```

Comment:
This reads really weird. `is_valid_utf16` isn't used for ensuring the string is the same in Rust, it's used for ignoring the string entirely.

Can we just link this somehow to `reference/types/string.md`? I think that's better than duplicating all that information (and possibly having it get out of sync, etc.)

Reply:
Agreed yeah, the hyperlink I think is the best way to go